controller/aks-cluster-config-handler.go

```diff
@@ -567,10 +567,9 @@ func (h *Handler) buildUpstreamClusterState(ctx context.Context, credentials *ak
 	upstreamSpec.KubernetesVersion = clusterState.KubernetesVersion
 
 	// set DNS prefix
-	if clusterState.DNSPrefix == nil {
-		return nil, fmt.Errorf("cannot detect cluster [%s] upstream DNS prefix", spec.ClusterName)
+	if clusterState.DNSPrefix != nil {
+		upstreamSpec.DNSPrefix = clusterState.DNSPrefix
```

Comment:
Not a blocker but a clarification question: what happens in case when `existing cluster DNSPrefix == nil`

Reply:
We would like to make possible only import AKS clusters with privateDNS zones.